src/include/storage/garbage_collector.h

```diff
@@ -0,0 +1,180 @@
+#pragma once
+
+#include <queue>
+#include <utility>
+#include "common/container/concurrent_queue.h"
+#include "common/macros.h"
+#include "loggers/storage_logger.h"
+#include "storage/data_table.h"
+#include "transaction/transaction_context.h"
+#include "transaction/transaction_manager.h"
+#include "transaction/transaction_util.h"
+
+namespace terrier::storage {
+
+/**
+ * The garbage collector is responsible for processing a queue of completed transactions from the transaction manager.
+ * Based on the contents of this queue, it unlinks the DeltaRecords from their version chains when no running
+ * transactions can view those versions anymore. It then stores those transactions to attempt to deallocate on the next
+ * iteration if no running transactions can still hold references to them.
+ */
+class GarbageCollector {
+ public:
+  GarbageCollector() = delete;
```

Comment:
don't need this delete, should be implicitly deleted after you declared a non-default constructor.